### ISSUE TITLE
Add support for x86-64 decompilation

### ISSUE THREAD

**Issue by radare** [nemerle/boomerang#49]

**Comment by hakusdream** [comment]
@nemerle Perhaps milestones would suit this better. Adding an architecture is several months in the making.

**Comment by nemerle** [comment]
Well, for now it's just a wish, since milestone would mean that we're actually going to implement it :smile:

**Comment by hakusdream** [comment]
True, although I wouldn't rule out someone committing a feature branch.

**Comment by hakusdream** [comment]
Note, the [AMD64 Architecture's Programming Manual](http://support.amd.com/TechDocs/24592.pdf) describes enough of the architecture to implement.

**Comment by nemerle** [comment]
The problem is not with the description of the AMD64, but with the cpu specification framework boomerang uses. It's a pretty obscure machine code parser generator using New Jersey Machine Code Toolkit. 
I've pondered generating boomerang SSL compatible asm statements using Capstone but it's waaaay down the priority list
